pkg/importer/imageio-datasource.go

```diff
@@ -277,30 +497,36 @@ func getTransfer(conn ConnectionInterface, diskID string) (*ovirtsdk4.ImageTrans
 		return nil, uint64(0), errors.New("Error disk id not available")
 	}
 
-	image, err := ovirtsdk4.NewImageBuilder().Id(id).Build()
-	if err != nil {
-		return nil, uint64(0), errors.Wrap(err, "Error building image object")
+	var imageTransferBuilder *ovirtsdk4.ImageTransferBuilder
+	if snapshot != nil {
+		imageTransferBuilder = ovirtsdk4.NewImageTransferBuilder().Snapshot(snapshot)
+	} else {
+		image, err := ovirtsdk4.NewImageBuilder().Id(id).Build()
+		if err != nil {
+			return nil, uint64(0), errors.Wrap(err, "Error building image object")
+		}
+		imageTransferBuilder = ovirtsdk4.NewImageTransferBuilder().Image(image)
 	}
 
-	transfersService := conn.SystemService().ImageTransfersService()
-	transfer := transfersService.Add()
-	imageTransfer, err := ovirtsdk4.NewImageTransferBuilder().Image(
-		image,
-	).Direction(
+	imageTransfer, err := imageTransferBuilder.Direction(
 		ovirtsdk4.IMAGETRANSFERDIRECTION_DOWNLOAD,
 	).Format(
-		ovirtsdk4.DISKFORMAT_RAW,
+		formatType,
+	).InactivityTimeout(
+		60,
 	).Build()
 	if err != nil {
 		return nil, uint64(0), errors.Wrap(err, "Error preparing transfer object")
 	}
 
+	transfersService := conn.SystemService().ImageTransfersService()
```

Comment:
Same question as before, can this be nil? if not why not just chain it? If it can be nil, we should check for it.

Reply:
Not as far as I know, unless `conn` itself is `nil`.

Reply:
I chained this one too.